src/libcrun/status.c

```diff
@@ -387,7 +387,30 @@ libcrun_read_container_status (libcrun_container_status_t *status, const char *s
 
   ret = read_all_file (file, &buffer, NULL, err);
   if (UNLIKELY (ret < 0))
-    return ret;
+    {
+
+      if (crun_error_get_errno (err) == ENOENT)
+        {
+          cleanup_free char *statedir = NULL;
+          libcrun_error_t tmp_err;
+          int tmp_ret;
+
+          tmp_ret = libcrun_get_state_directory (&statedir, state_root, id, &tmp_err);
+          if (UNLIKELY (tmp_ret < 0))
+            crun_error_release (&tmp_err);
+          else
+            {
+              tmp_ret = crun_path_exists (statedir, &tmp_err);
+              if (UNLIKELY (tmp_ret < 0))
+                crun_error_release (&tmp_err);
+              else if (tmp_ret == 0)
+                return crun_error_wrap (err, "container `%s` does not exist", id);
+            }
+          return ret;
```

Comment:
This line could be removed

Reply:
we still need it in case we take the `if` branch.  If it is not clear, should I move the return into the `if` branch?

Reply:
I thought line 412 takes care of returning `ret`

Reply:
you are right.  Pushed a fixed version.